test/realtime/message.test.js

```diff
@@ -917,6 +917,75 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
       );
     });
 
+    it('subscribe_with_filter_object', function (done) {
+      const realtime = helper.AblyRealtime();
+      const channel = realtime.channels.get('subscribe_with_filter_array');
+
+      function send(cb) {
+        channel.publish(
+          [
+            {
+              name: 'correct',
+              extras: {
+                reference: {
+                  type: 'com.ably.test',
+                  id: '0123456789',
+                },
+              },
+            },
+            {
+              name: 'incorrect-noref',
+            },
+            {
+              name: 'incorrect-badtype',
+              extras: {
+                reference: {
+                  type: 'com.ably.incorrect',
+                  id: '0123456789',
+                },
+              },
+            },
+            {
+              name: 'incorrect-badid',
+              extras: {
+                reference: {
+                  type: 'com.ably.test',
+                  id: '000000000000',
+                },
+              },
+            },
+          ],
+          cb
+        );
+      }
+
+      function subscribe(cb) {
+        channel.subscribe(
+          {
+            refType: 'com.ably.test',
+            refId: '0123456789',
+          },
+          (m) => {
+            try {
+              expect(m.name).to.be.equal('correct', 'Correct message received');
+            } catch (e) {
+              return cb(e);
+            }
+            config.nextTick(cb);
```

Comment:
I'd like to see more comments in this fixture, in particular the purpose of this line is non-obvious; what is it that we're expecting to happen before the next tick?

Reply:
[Done](https://github.com/ably/ably-js/pull/1003/commits/3192a0bdede12e59e37edb4eda726dcf8e708f75)